src/QueryLanguage/Walker/Dbal/SqlWalker.php

```diff
@@ -0,0 +1,153 @@
+<?php declare(strict_types=1);
+
+namespace Fazland\ApiPlatformBundle\QueryLanguage\Walker\Dbal;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Query\QueryBuilder;
+use Doctrine\DBAL\Types\Types;
+use Fazland\ApiPlatformBundle\QueryLanguage\Expression\ExpressionInterface;
+use Fazland\ApiPlatformBundle\QueryLanguage\Expression\Literal\LiteralExpression;
+use Fazland\ApiPlatformBundle\QueryLanguage\Expression\Literal\NullExpression;
+use Fazland\ApiPlatformBundle\QueryLanguage\Expression\Literal\StringExpression;
+use Fazland\ApiPlatformBundle\QueryLanguage\Expression\ValueExpression;
+use Fazland\ApiPlatformBundle\QueryLanguage\Walker\AbstractWalker;
+
+class SqlWalker extends AbstractWalker
+{
+    private AbstractPlatform $platform;
+    protected QueryBuilder $queryBuilder;
+    private ?string $columnType;
+    private string $fieldName;
+
+    public function __construct(QueryBuilder $queryBuilder, string $field, ?string $columnType = null)
+    {
+        $this->platform = $queryBuilder->getConnection()->getDatabasePlatform();
+        parent::__construct($this->platform->quoteIdentifier($field));
+
+        $this->fieldName = $field;
+        $this->queryBuilder = $queryBuilder;
+        $this->columnType = $columnType;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function walkLiteral(LiteralExpression $expression)
+    {
+        $value = parent::walkLiteral($expression);
+        if ($expression instanceof NullExpression) {
+            return $value;
+        }
+
+        switch ($this->columnType) {
+            case Types::DATETIME_MUTABLE:
+            case Types::DATETIMETZ_MUTABLE:
+                return new \DateTime($value);
+
+            case Types::DATETIME_IMMUTABLE:
+            case Types::DATETIMETZ_IMMUTABLE:
+                return new \DateTimeImmutable($value);
+
+            default:
+                return $value;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function walkComparison(string $operator, ValueExpression $expression)
+    {
+        $field = $this->field;
+        if ($expression instanceof NullExpression) {
+            return $field.'IS NULL';
```

Comment:
Shouldn't be a white space being prefixed of `IS NULL`?
Something like `return $field.' IS NULL';`

Reply:
Good catch